### PR TITLE
Add rego sample

### PR DESCRIPTION
**Pull request by mdanielolsson**
This commit adds an example .rego for domain based usernames/topic names. A new directory for examples have been added so we can have multiple different packages for different use cases.

### CODE REVIEW

File: src/main/rego/domain_based_policy/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Policy sample
+
+This sample is meant for more dynamic Kafka environments where topics comes and goes.
```
Comment (lukasz-kaminski) [inline src/main/rego/domain_based_policy/README.md]: `topics comes and goes` -> `topics come and go`

File: src/main/rego/domain_based_policy/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+It got a standard policy that can be used by following a domain based naming convention on topics and users, but also supports custom data to extend this policy with whitelisting and blacklisting.
```
Comment (lukasz-kaminski) [inline src/main/rego/domain_based_policy/README.md]: `It has got`

File: src/main/rego/domain_based_policy/policy.rego [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	not blacklisted_produce
+	whitelisted_producer
+}
+
+# --------------------- FUNCTIONS ---------------------- #
+
+inter_broker_communication {
+    # If inter-broker communication is unauthenticated
+    # they will come with username ANONYMOUS. Make sure to protect
+    # this using network rules, or concider having authenticated
```
Comment (lukasz-kaminski) [inline src/main/rego/domain_based_policy/policy.rego]: `concider` -> `consider`

File: src/test/rego/domain_based_policy/test_data/data.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            ],
+            "denied_consume": [
+                "xyz"
+            ],
+            "denied_produce": [
+                "xyz"
+            ]
+        }
+    }
+}
```
Comment (lukasz-kaminski) [inline src/test/rego/domain_based_policy/test_data/data.json]: No newline at the end of file.

**Comment by lukasz-kaminski** [comment]
Your build is failing because in `domain_based_policy/policy.rego` and `owner_based_policy/policy.rego` you've declared the same package. Both of the files start with `default allow ...` which essentially means defining default value twice, which is unacceptable to OPA.

My initial advice was to rename packages to more specific ones, but I'm not sure if both of policies should coexist or not.

**Comment by mdanielolsson** [comment]
> Your build is failing because in `domain_based_policy/policy.rego` and `owner_based_policy/policy.rego` you've declared the same package. Both of the files start with `default allow ...` which essentially means defining default value twice, which is unacceptable to OPA.
> 
> My initial advice was to rename packages to more specific ones, but I'm not sure if both of policies should coexist or not.

It's just a different example and I don't think it hurt if they coexist but only one package can be used in the Kafka plugin so maybe it doesn't make sense.

I think we should only have the original rego here and then have a set of other examples somewhere else and only run opa tests on them.. I'll see what I can do.

**Comment by mdanielolsson** [comment]
Updated the PR and moved the example to a new examples directory where we can store example rego packages and other examples.

**Pull request merged**